### PR TITLE
ref(alerts): Add support for dynamic crash rate alert

### PR DESCRIPTION
**Pull request by ceorourke**
Add support for a crash rate alert using anomaly detection. This requires a different snuba query.

Closes https://getsentry.atlassian.net/browse/ALRT-240

### CODE REVIEW

File: src/sentry/seer/anomaly_detection/store_data.py [hunk 1]
```diff
@@ -154,17 +179,54 @@ def fetch_historical_data(
...
-            organization=alert_rule.organization,
-            projects=[project],
-            start=start,
-            end=end,
-        ),
-        rollup=granularity,
-        referrer=Referrer.ANOMALY_DETECTION_HISTORICAL_DATA_QUERY.value,
-        zerofill_results=True,
-    )
+    if dataset == metrics_performance:
```
Comment (ceorourke) [inline src/sentry/seer/anomaly_detection/store_data.py:182]: This was all pulled from https://github.com/getsentry/sentry/blob/master/src/sentry/api/endpoints/organization_sessions.py

Comment (nhsiehgit) [reply src/sentry/seer/anomaly_detection/store_data.py:182]: would be great if we could separate concerns here a bit and pull this into some helpers?

Maybe add a note or create a ticket to do this as part of https://getsentry.atlassian.net/browse/ALRT-187?

**Review by nhsiehgit — APPROVED** [review]
Would prefer if we could find a way to decouple these concerns and dedup the logic 😭 

but if this passes, then we can do those as follow up items and get these in to unblock

File: src/sentry/seer/anomaly_detection/store_data.py (hunk 1 above)
Comment (ceorourke) [reply src/sentry/seer/anomaly_detection/store_data.py:225]: updated to add `get_crash_free_historical_data()`

**Pull request merged**